core/src/test/scala/com/yahoo/maha/core/query/DefaultQueryPipelineFactoryTest.scala

```diff
@@ -501,17 +501,23 @@ class DefaultQueryPipelineFactoryTest extends FunSuite with Matchers with Before
         row.addValue("Ad Group ID", 202)
         row.addValue("Impressions", 100)
         row.addValue("Clicks", 1)
+        row.addValue("Pricing Type", "CPC")
         rl.addRow(row)
+
+        val row2 = rl.newRow
+        row2.addValue("Campaign ID", 15)
+        row2.addValue("Ad Group ID", 202)
+        row2.addValue("Impressions", 100)
+        row2.addValue("Clicks", 1)
+        row2.addValue("Pricing Type", "CPA")
+        rl.addRow(row2)
     }.withOracleCallback {
       rl =>
         val row = rl.newRow
         row.addValue("Advertiser ID", 1)
         row.addValue("Ad Group Status", "ON")
         row.addValue("Campaign ID", 10)
         row.addValue("Ad Group ID", 202)
-        row.addValue("Source", 2)
```

Comment:
This test is running a dim driven query.  No reason to modify this test.  It works as expected.

Reply:
This test is getting a Row with Pricing Type, Destination URL, and Source from a Dimension that does not contain any of those keys.  The returned Row List was invalid in the first place.

Reply:
More concretely,
`println(pipeline.queryChain.subsequentQueryList.head.asInstanceOf[OracleQuery].asString)`

Prints
`(SELECT * FROM (SELECT D.*, ROWNUM AS ROW_NUMBER FROM (SELECT  *
      FROM (SELECT ago0.advertiser_id "Advertiser ID", ago0."Ad Group Status" "Ad Group Status", ago0.id "Ad Group ID"
            FROM 
                (SELECT  DECODE(status, 'ON', 'ON', 'OFF') AS "Ad Group Status", advertiser_id, id
            FROM ad_group_oracle
            WHERE (advertiser_id = 213) AND (id IN (10))
             ) ago0 
           

           )
            ) D )) UNION ALL (SELECT * FROM (SELECT D.*, ROWNUM AS ROW_NUMBER FROM (SELECT * FROM (SELECT  *
      FROM (SELECT ago0.advertiser_id "Advertiser ID", ago0."Ad Group Status" "Ad Group Status", ago0.id "Ad Group ID"
            FROM 
                (SELECT  DECODE(status, 'ON', 'ON', 'OFF') AS "Ad Group Status", advertiser_id, id
            FROM ad_group_oracle
            WHERE (advertiser_id = 213) AND (id NOT IN (10))
             ) ago0 
           

           )
            ) WHERE ROWNUM <= 100) D ) WHERE ROW_NUMBER >= 1 AND ROW_NUMBER <= 100)`

Reply:
The point is, it is hard to verify your changes and review them if you keep modifying tests that have nothing to do with new functionality.  As I noted in another comment, keep bug fixes or test fixes separate from new functionality.  What you are doing just makes it harder to review your code and makes it difficult to be confident about the changes not breaking other functionality.